### PR TITLE
feat: full sensor data, restart connection/router

### PR DESCRIPTION
**Pull request by chemelli74**
Added:

- restart_connection()
- restart_router()
- full sensor data retrival (uptime, serial #, hardware release, etc.)
- 
Ref #7

### CODE REVIEW

File: src/aiovodafone/api.py [hunk 1]
```diff
@@ -209,19 +210,28 @@ async def _get_page_result(self, page: str) -> dict[Any, Any]:
...
-    async def get_user_data(self) -> dict[Any, Any]:
+    async def get_sensor_data(self) -> dict[Any, Any]:
         """Load user_data page information."""
         _LOGGER.debug("Getting sensor data for host %s", self.host)
 
-        reply_dict = await self._get_page_result("/data/user_data.json")
-        return reply_dict
+        reply_dict_1 = await self._get_page_result("/data/user_data.json")
+        reply_dict_2 = await self._get_page_result("/data/statussupportstatus.json")
+        reply_dict_3 = await self._get_page_result("/data/statussupportrestart.json")
```
Comment (bdraco) [inline src/aiovodafone/api.py:219]: Merging the data together seems like it makes it harder to handle.

If you want to return it in a single call, I would wrap it in a `dataclass` with 4 values

File: src/aiovodafone/api.py [hunk 2]
```diff
@@ -264,6 +274,14 @@ async def get_all_devices(self) -> dict[str, VodafoneStationDevice]:
 
         return self._devices
 
+    async def convert_uptime(self, uptime: str) -> datetime:
+        """Convert router uptime to last boot datetime."""
+        d = int(uptime.split(":")[0])
+        h = int(uptime.split(":")[1])
+        m = int(uptime.split(":")[2])
+
+        return datetime.utcnow() - timedelta(days=d, hours=h, minutes=m)
```
Comment (bdraco) [inline src/aiovodafone/api.py:283]: You might be able to use `isodata.parse_duration` here .. but what you have is probably fine for this usecase

File: src/aiovodafone/api.py (hunk 1 above)
Comment (chemelli74) [reply src/aiovodafone/api.py:219]: Each call returns a `dict` with few values.
Using such single `dict` in HomeAssistant integration makes it easy to access info without the complexity to know where those are originally stored.

Simone

Comment (bdraco) [reply src/aiovodafone/api.py:219]: From an HA perspective, how you implement in the lib doesn't really matter so if thats the way you want to do it that is OK

**Pull request merged**